src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -412,6 +412,9 @@ private void HandleField(IComplexGraphType parentType, FieldType field, TypeColl
 
             if (field.ResolvedType == null)
             {
+                if (field.Type == null)
+                    throw new InvalidOperationException($"Both ResolvedType and Type properties on field '{parentType?.Name}.{field.Name}' are null.");
```

Comment:
While fixing the tests, I discovered that NRE was happening a bit later in `AddTypeIfNotRegistered`